cm2350/e200z7.py

```diff
@@ -224,8 +224,9 @@ def shutdown(self):
             # Go through each peripheral and if any of them have a server thread
             # running, stop it now
             for mname in list(self.modules):
-                if hasattr(self.modules[mname], 'stop'):
-                    self.modules[mname].stop()
+                if hasattr(self.modules[mname], 'shutdown'):
+                    logger.debug("shutdown: Cleaning up %s...", mname)
+                    self.modules[mname].shutdown()
```

Comment:
As a nitpick, for aesthetics, it's probably just best to have the base module have:
```
def shutdown(self):
    pass
```
and skip the `hasattr` checks

Reply:
That's a good point, the base class should have that now. The only case is if someone doesn't use the base `Peripheral` class, then they won't get the base init, reset, and shutdown functions.  The `e200GDB` class is such an example. so if those functions aren't in there by default then the core shutdown function may raise an exception unless we handle it gracefully

Reply:
We should mandate that all modules derive from some base class that "implements" all the API required (loosely defined, meaning: raising exceptions for unimplemented functions that should be overriden, and `pass` for ones that are ok to omit)

Actually, `Peripheral` should subclass this base class as well, as should `e200GDB`